internal/node/machine_mgr.go

```diff
@@ -131,17 +133,11 @@ func (m *MachineManager) Start() error {
 	return nil
 }
 
-func (m *MachineManager) DeployWorkload(vm *runningFirecracker, workloadName, namespace string, request controlapi.RunRequest) error {
-	// TODO: make the bytes and hash/digest available to the agent
-	req := agentapi.DeployRequest{
-		WorkloadName:    &workloadName,
-		Hash:            nil, // FIXME
-		TotalBytes:      nil, // FIXME
-		TriggerSubjects: request.TriggerSubjects,
-		WorkloadType:    request.WorkloadType, // FIXME-- audit all types for string -> *string, and validate...
-		Environment:     request.WorkloadEnvironment,
+func (m *MachineManager) DeployWorkload(vm *runningFirecracker, runRequest controlapi.RunRequest, deployRequest agentapi.DeployRequest) error {
```

Comment:
It seems weird that we pass two different requests for `Deploy`. It was already weird that we were constructing a `DeployRequest` in this method, using mostly identically-named fields from the given `RunRequest`.

Reply:
I agree.  I was going to try and fix that in another PR so this one didn't start to balloon.  I need to create a issue for it

Reply:
On board with that.